File: 002/kmats/002.rb
def rec a, b
  if a == 1 && b == 2
    b + rec(b, a + b)
  elsif a + b > 4000000
    0
  else
    if (a + b) % 2 == 0
      a + b + rec(b, a + b)
    else
      rec(b, a + b)
    end
  end
end

p rec 1, 2

File: 002/kmats/002.exs
defmodule Problem002 do
  def rec(a, b) do
    cond do
      a == 1 and b == 2 ->
        b + rec(b, a + b)
      a + b > 4_000_000 ->
        0
      true ->
        if rem(a + b, 2) == 0 do
          a + b + rec(b, a + b)
        else
          rec(b, a + b)
        end
    end
  end
end

IO.puts Problem002.rec(1, 2)
